spark/src/main/scala/geotrellis/spark/RasterMetaData.scala

```diff
@@ -2,26 +2,37 @@ package geotrellis.spark
 
 import geotrellis.raster._
 import geotrellis.spark.tiling._
-import geotrellis.vector.Extent
+import geotrellis.vector.{ProjectedExtent, Extent}
 
 import geotrellis.proj4.CRS
 
 import org.apache.spark.rdd._
 
+/**
+ * @param cellType    value type of each cell
+ * @param layout      definition of the tiled raster layout
+ * @param dataExtent  Extent covering the source data
+ * @param crs         CRS of the raster projection
+ */
 case class RasterMetaData(
   cellType: CellType,
-  extent: Extent,
-  crs: CRS,
-  tileLayout: TileLayout
+  layout: LayoutDefinition,
+  dataExtent: Extent,
```

Comment:
Why does this need to be `dataExtent` rather than `extent`? I think with the `LayoutDefinition` (which the member name should probably be the more verbose `layoutDefinition` to not confuse it with `TileLayout`.